sky/data/data_utils.py

```diff
@@ -126,6 +146,175 @@ def verify_gcs_bucket(name: str) -> bool:
         return False
 
 
+def create_az_client(client_type: str,
+                     storage_account_name: Optional[str] = None,
+                     container_name: Optional[str] = None) -> Client:
+    """Helper method that connects to AZ client for diverse Resources.
+
+    Args:
+      type: str; resource name, e.g. storage, resource, container
+      storage_account_name: str; name of the storage account
+      container_name: str; name of the container.
+
+    Returns:
+      Client object facing AZ Resource of the 'type'.
+    """
+    if client_type == 'container':
+        assert storage_account_name is not None
+        assert container_name is not None
+    subscription_id = azure.get_subscription_id()
+    return azure.get_client(client_type, subscription_id, storage_account_name,
+                            container_name)
+
+
+def verify_az_bucket(storage_account_name: str, container_name: str) -> bool:
+    """Helper method that checks if the AZ Container exists
+
+    Args:
+      name: str; Name of AZ Container.
+
+    Returns:
+      boolean; shows either or not the container exists.
+    """
+    container_client = create_az_client('container', storage_account_name,
+                                        container_name)
+    return container_client.exists()
+
+
+def get_az_resource_group(
+        storage_account_name: str,
+        storage_client: Optional[Client] = None) -> Optional[str]:
+    """Returns the resource group name the given storage account belongs to.
+
+    Args:
+        storage_account_name: str; name of the storage account
+        storage_client: Optional[Client]; client object facing storage
+
+    Returns:
+        Name of the resource group the given storage account belongs to.
+    """
+    if storage_client is None:
+        storage_client = create_az_client('storage')
+    for account in storage_client.storage_accounts.list():
+        if account.name == storage_account_name:
+            # Extract the resource group name from the account ID
+            resource_group_name = account.id.split('/')[4]
```

Comment:
`account.id` has a fixed format of an output as a result of `storage_client.storage_accounts.list()` rather than it being a variable output depending on user input. Not sure if it's absolutely necessary to error handle as it'll always have a fixed length?

I added an example of account.id as a comment.

Reply:
If it is guaranteed to be a fixed length, could we add an assertion here?

Reply:
Done.